### DISCUSSION TITLE
Give to ebsd.orientations the orientation of grains.meanOrientation

### DISCUSSION THREAD

**Discussion by Fred-Ad** [mtex-toolbox/mtex#1198] [Ask Anything] [answered]
Hi

I want to give to `ebsd.orientations`, the values of `grains.meanOrientation`.
I can do it like this : 

```
ori = grains.meanOrientation

for i = 1 : size(grains)
  ebsd(ebsd.grainId == i).orientations = ori(i);
end
```

But for a 600000points map, it's very long

I didn't find a matrix way to make the calculation. Any idea ?
 
thanks a lot

Fred

**Comment by zmichels** [comment]
You could try something like the following:ebsd.prop.grainMean = grains(ebsd.grainId).meanRotation Or you could loop through the different phases (rather than each grain in your example), and do something like the following to get a crystal orientation rather than a rotation:  % get unique mineral namesminerals = ebsd.mineralList; % loopfor n = 1:length(minerals)       ebsd(minerals{n}).prop.grainMeanO = grains('id',ebsd(minerals{n}).grainId).meanOrientation;end I’d be interested to know if that works for you. Cheers,Zach From: Fred-AdSent: Monday, September 13, 2021 1:55 PMTo: mtex-toolbox/mtexCc: SubscribedSubject: [mtex-toolbox/mtex] Give to ebsd.orientations the orientation of grains.meanOrientation (Discussion #1198) HiI want to give to ebsd.orientations, the values of grains.meanOrientation.I can do it like this :ori = grains.meanOrientation for i = 1 : size(grains)  ebsd(ebsd.grainId == i).orientations = ori(i);endBut for a 600000points map, it's very longI didn't find a matrix way to make the calculation. Any idea ?thanks a lotFred—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by nyyssont** [comment, marked as answer]
You could try the following:

```
ebsd('Your phase').orientations = grains(ebsd('Your phase').grainId).meanOrientation
```

Repeat as necessary for various phases.

**Comment by Fred-Ad** [comment]
I've chose the Tuomo's one, very easy in fact. Work like a charm.

Thanks a lot, both Zach and Tuomo !
